cmd/virt-handler/virt-handler.go

```diff
@@ -113,62 +113,51 @@ var _ service.Service = &virtHandlerApp{}
 func (app *virtHandlerApp) getSelfSignedCert() error {
 	var ok bool
 
-	generateCerts := false
-	secret, err := app.virtCli.CoreV1().Secrets(app.namespace).Get(virtHandlerCertSecretName, metav1.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			generateCerts = true
-		} else {
-			return err
-		}
-	}
-
-	if generateCerts {
-		// Generate new certs if secret doesn't already exist
-		caKeyPair, _ := triple.NewCA("kubevirt.io")
-
-		clientKeyPair, _ := triple.NewClientKeyPair(caKeyPair,
-			"kubevirt.io:system:node:virt-handler",
-			nil,
-		)
-
-		app.clientKeyBytes = cert.EncodePrivateKeyPEM(clientKeyPair.Key)
-		app.clientCertBytes = cert.EncodeCertPEM(clientKeyPair.Cert)
-		app.signingCertBytes = cert.EncodeCertPEM(caKeyPair.Cert)
-
-		secret := k8sv1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      virtHandlerCertSecretName,
-				Namespace: app.namespace,
-				Labels: map[string]string{
-					v1.AppLabel: "virt-api-aggregator",
-				},
-			},
-			Type: "Opaque",
-			Data: map[string][]byte{
-				clientCertBytesValue:  app.clientCertBytes,
-				clientKeyBytesValue:   app.clientKeyBytes,
-				signingCertBytesValue: app.signingCertBytes,
+	caKeyPair, _ := triple.NewCA("kubevirt.io")
+	clientKeyPair, _ := triple.NewClientKeyPair(caKeyPair,
+		"kubevirt.io:system:node:virt-handler",
+		nil,
+	)
+	app.clientKeyBytes = cert.EncodePrivateKeyPEM(clientKeyPair.Key)
```

Comment:
can we remove setting these app cert values from the local generated cert object and always set them directly from the secret?

The code is technically correct the way you have it, but my concern is that it leaves room to be misinterpreted in the future.

Reply:
Done.